packages/renderer/src/index.js

```diff
@@ -45,7 +45,7 @@ const pdf = initialValue => {
       pageMode,
     });
 
-    const layout = await layoutDocument(container.document, fontStore);
+    const layout = await layoutDocument(container.document, fontStore, currentPageCallBack);
```

Comment:
This seems very specific to this case. Seems likely we want to report more things to the outside. Maybe we can pass an event emitter to layout, and internally emit events?

Reply:
call back looks fairly simple and straightforward, but as you are saying to pass an event to layout, when i was doing the call back, i tried to add the calllback to all the 12 functions, but it was slowing the speed of pdf creation, so i just added them to few, worried about the performance overhead if we implement in all the steps,

emitting events also makes senes to me, both works!